### PR TITLE
[HIP] Changes for device_globals and enable tests

### PR DESCRIPTION
**Pull request by hdelan**
Enable `device_global` for AMDGPU. Related to https://github.com/oneapi-src/unified-runtime/pull/1186 .

In order for `hipModuleGetGlobal` to see a global symbol we need to not make the visibility of the symbol hidden. Perhaps changing this as a default is not always a good idea. Could also do this in the SYCL headers using an attribute to specify the visibility of the `device_global` var

### CODE REVIEW

**Review by steffenlarsen — APPROVED** [review]
Very nice! I'm glad to see the program-metadata approach not being too much of a pain to include into the HIP backend. Thank you, @hdelan !

**Comment by jinz2014** [comment]
I have a question. Does hipcc see a global symbol by making visibility of symbols ?  Thanks.

**Comment by hdelan** [comment]
> I have a question. Does hipcc see a global symbol by making visibility of symbols ? Thanks.

Hi @jinz2014 it seems that HIPCC gives device globals `protected` visibility in the resulting binary.

**Comment by aarongreig** [comment]
update the tag and we're good to go @hdelan

**Comment by hdelan** [comment]
> update the tag and we're good to go @hdelan

Thanks @aarongreig ! Have updated

**Comment by aarongreig** [comment]
ah might have been too quick on the draw from https://github.com/intel/llvm/pull/12381 merging you'll need to resolve the conflict

**Comment by hdelan** [comment]
> ah might have been too quick on the draw from #12381 merging you'll need to resolve the conflict

Yep just resolved

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -56,15 +56,13 @@ endif()
...
-  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
-  # commit d06ba9d9c539dc712444c6db95dfe1629bd5f7d8
-  # Merge:  e1414e1 35b6a5e
-  # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
-  # Date:   Thu Jan 11 15:09:01 2024 +0000
-  #     Merge pull request #1035 from Bensuo/cmd-buffer-profiling-l0
-  #     [EXP][CMDBUF] Add support for recovering CommandBuffer profiling
-  #     information
-  set(UNIFIED_RUNTIME_TAG d06ba9d9c539dc712444c6db95dfe1629bd5f7d8)
+  set(UNIFIED_RUNTIME_REPO https://github.com/oneapi-src/unified-runtime.git)
```
Comment (aarongreig) [inline sycl/plugins/unified_runtime/CMakeLists.txt:59]: not sure if there's actually a semantic difference dropping the quotes but it does make me a bit nervous

Comment (hdelan) [reply sycl/plugins/unified_runtime/CMakeLists.txt:59]: Aha didn't cop this. Should be the same but I will change back

**Review by aarongreig — APPROVED** [review]

**Comment by kbenzie** [comment]
@intel/dpcpp-clang-driver-reviewers please review

**Comment by hdelan** [comment]
@steffenlarsen we are blocked here awaiting @intel/dpcpp-clang-driver-reviewers . Quite a small group and it seems they are off atm. What should we do here?

**Comment by steffenlarsen** [comment]
I believe they should be back today. Tag @mdtoguchi.

If I am wrong, I think we can merge and allow for a post-commit review if needed, though the driver changes don't seem so intrusive to me. Let's give it another day though.

**Review by mdtoguchi — COMMENTED** [review]
Driver changes are OK - could you add some LIT testing to verify the additional option passing to `sycl-post-link` and compile step?

**Comment by hdelan** [comment]
Thanks @mdtoguchi have added a test for driver

**Review by mdtoguchi — APPROVED** [review]
LGTM - thanks!

**Comment by kbenzie** [comment]
@intel/llvm-gatekeepers please merge

**Pull request merged**